usagov/usagov-2021
# USAGOV-1945: Retrieve call center EWT direct from s3 bucket

## Jira Task


https://cm-jira.usa.gov/browse/USAGOV-1945

## Description

- cron app: 
  - Provide contact center API variables to cron app
  - Update cron app to call contact center API conditionally (on CALL_CENTER_RUN) variable
  - Modifies location of waittime.json file written by cron app
- cms app: 
  - Retrieve waittime.json direct from s3 bucket (hard-coded to prod bucket)
  - check timestamp and reject outdated waittime.json values
  - remove route to wait-time, since we can get it direct from s3 

## Type of Changes

  - Add screenshot showing what it should look like

## Testing Instructions



### Change Requirements



### Validation Steps

I've set this up so the cron app will run the call-center-update script only in prod, by default. (`bin/cloudgov/deploy-cron` sets CALL_CENTER_RUN to "true" only for prod.) To get it to run from another space, one would run: 
```
cf set-env cron CALL_CENTER_RUN "true"
cf restage cron
```

**Cron app changes:** I deployed the cron app to the `dr` space and ran the incantation above. After it ran for a little bit, I stopped the app, but you can simply do `cf target -s dr; cf start cron` if you want to watch it go or get updated data. 

The resulting waittime.json file can be retrieved from https://s3-us-gov-west-1.amazonaws.com/cg-fa02c8c6-1731-4c83-8199-d96e3c21f125/1.0/waittime.json

(Note that the s3 bucket in that URL is the one for the `dr` space, not the one for `prod`. 

**CMS app changes:** IMHO this could wait, beyond verifying that the CMS still builds and runs. :-) Until this makes it to production, it will be necessary to edit the bucket name in web/themes/custom/usagov/scripts/callCenterWaitTimeLookup.js; you can use cg-fa02c8c6-1731-4c83-8199-d96e3c21f125 . Put a div with the id "callCenterTime" on a page (the /phone page is the expected location). The javascript is set up to populate that only if there are wait times; outside of working hours the API will return "-1" and we just want to be silent. So this will work best if it's running live. I don't know whether we've nailed down how this should be styled (or even if it should be a div) -- @jacobaaronyeager might know that. 

**Note:** I ran the code scans. The failure isn't due to any of these changes, and it's already logged in Jira. 

## Security Review


- [x] Requires SSPP updates - maybe? I think we should mention the call center API if we don't do that already.


## Reviewer Reminders

- Reviewed code changes
- Reviewed functionality
- Security review complete or not required

## Post PR Approval Instructions

Follow these steps as soon as you merge the new changes.

1. Go to the [USAGov Circle CI project](https://app.circleci.com/pipelines/github/usagov/usagov-2021).
2. Find the commit of this pull request.
3. Build and deploy the changes.
4. Update the Jira ticket by changing the ticket status to `Review in Test` and add a comment. State whether the change is already visible on [cms-dev.usa.gov](http://cms-dev.usa.gov/) and [beta-dev.usa.gov](http://beta-dev.usa.gov/), or if the deployment is still in process.
